src/account/account.service.ts

```diff
@@ -633,6 +633,41 @@ export class AccountService {
         await this.accountRepository.save(updated);
     }
 
+    async updateAccountByApId(
+        apId: URL,
+        data: {
+            name: string | null;
+            bio: string | null;
+            username: string;
+            avatarUrl: string | null;
+            bannerImageUrl: string | null;
+            url: string | null;
+        },
+    ): Promise<Result<true, 'account-not-found'>> {
```

Comment:
I think generally we return the updated entity from the service, but not a big deal